SCons/Scanner/__init__.py

```diff
@@ -330,24 +346,26 @@ def _scan(node, _, path=(), self=self):
                 return []
             return self.scan(node, path)
 
-        kw['function'] = _scan
-        kw['path_function'] = FindPathDirs(path_variable)
+        kwargs['function'] = _scan
+        kwargs['path_function'] = FindPathDirs(path_variable)
 
         # Allow recursive to propagate if child class specifies.
         # In this case resource scanner needs to specify a filter on which files
         # get recursively processed.  Previously was hardcoded to 1 instead of
         # defaulted to 1.
-        kw['recursive'] = kw.get('recursive', 1)
-        kw['skeys'] = suffixes
-        kw['name'] = name
+        kwargs['recursive'] = kwargs.get('recursive', 1)
```

Comment:
Could this be `True` instead of `1` or is `recursive` not-boolean?

Reply:
Pretty sure it can. I've usually made that change when I'm touching anyway, but didn't think of it here.